src/cms/utils/slug_utils.py

```diff
@@ -26,76 +61,77 @@ def generate_unique_slug(form_object, foreign_model):
     * :func:`cms.forms.events.event_translation_form.EventTranslationForm.clean_slug`
     * :func:`cms.forms.pois.poi_translation_form.POITranslationForm.clean_slug`
 
-    :param form_object: The form which contains the slug field
-    :type form_object: ~django.forms.Form
+    :param args: The supplied arguments list
+    :type args: list
 
-    :param foreign_model: If the form instance has a foreign key to another model (e.g. because it is a translation of
-                          a content-object), this paramaeter contains the model of the foreign related object.
-    :type foreign_model: ~django.db.models.Model
+    :param kwargs: The supplied keyword arguments
+    :type kwargs: dict
 
     :raises ~django.core.exceptions.ValidationError: When no slug is given and there is also no field which can be used
                                                      as fallback (either ``title`` or ``name``).
 
     :return: An unique slug identifier
     :rtype: str
     """
+    slug = kwargs["slug"]
+    foreign_model = kwargs["foreign_model"]
+    object_instance = kwargs["object_instance"]
+    fallback = kwargs["fallback"]
+    if args:
+        region = args[0]
+        language = args[1]
 
     logger.info("generate_unique_slug()")
     logger.info('foreign_model: "%s"', foreign_model)
     if foreign_model in ["page", "event", "poi"]:
-        logger.info('region: "%s"', form_object.region)
-        logger.info('language: "%s"', form_object.language)
-
-    slug = form_object.cleaned_data["slug"]
-
-    # if slug is empty, generate from title/name
-    if not slug:
-        # determine fallback field of the model
-        if foreign_model in ["page", "event", "poi"]:
-            fallback = "title"
-        else:
-            fallback = "name"
-        if fallback not in form_object.cleaned_data:
+        logger.info('region: "%s"', region)
+        logger.info('language: "%s"', language)
+
+    # if slug is empty and fallback field is set, generate from fallback:title/name
+    if not slug and fallback:
+        if not hasattr(object_instance, fallback):
```

Comment:
```suggestion
        if fallback not in cleaned_data:
```
Now that you're not accessing the fallback-field of the `object_instance`, but the cleaned data instead via `kwargs["cleaned_data"][fallback]`, you also need to check whether the value is present in the `cleaned_data` instead of the `object_instance`...

Reply:
Good point, I forgot to revert the if statement as well